hoomd/data/data_structures.py

```diff
@@ -0,0 +1,497 @@
+from abc import ABCMeta, abstractmethod
+from collections.abc import MutableMapping, MutableSequence, MutableSet
+from contextlib import contextmanager
+from copy import deepcopy
+from itertools import cycle
+
+from hoomd.data.typeconverter import (
+    TypeConversionError, TypeConverterValue, OnlyIf, Either,
+    TypeConverterMapping, TypeConverterSequence)
+
+
+class _HOOMDDataStructures(metaclass=ABCMeta):
```

Comment:
I recommend avoiding HOOMD in the class name. Also this should be singular (Structure, not Structures).
```suggestion
class _SyncedDataStructure(metaclass=ABCMeta):
```